### DISCUSSION TITLE
Question to available draid pool size

### DISCUSSION THREAD

**Discussion by ChristophSchmidpeter** [openzfs/zfs#16365] [Q&A] [answered]
I created a dRAID pool via `draid1:1d:3c:0s-0` on 3 drives with 3TB each.

For `zpool list` I would expect 15TiB*(1000/1024)^4=13.64TiB, which matches the output:
```
zpool list
NAME         SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
ZfsBackup1  13.6T   792G  12.9T        -         -     0%     5%  1.00x  DEGRADED  -
```
To my understanding, `draid1:1d:3c:0s-0` should create a pool with 1x redundancy, and thus
I would expect available storage to user to be (3-1)*5TiB*(1000/1024)^4= ~9.09TiB.
However, `zfs list` outputs only ~6.7 TiB:
```
zfs list:
NAME                                                USED  AVAIL  REFER  MOUNTPOINT
ZfsBackup1                                          396G  6.31T   192K  /ZfsBackup1
```

Why does zfs list show only ~6.7TB of storage and not ~9.09TiB of available storage? Is this expected behaviour or a bug?

**Comment by ChristophSchmidpeter** [comment]
@scineram Why the thumbs down emoji?

**Comment by amotin** [comment, marked as answer]
I think when you specified `1d` in `draid1:1d:3c:0s` you've effectively made it to add parity disk to each one data disk, so you effectively get a crippled form of mirror with 100% redundancy, that is why you get only a half capacity.  Proper 3-disk draid, if there is such a thing, would be `draid1:2d:3c:0s`, which is what would probably happen if you just specify `draid`.

**Comment by ChristophSchmidpeter** [comment]
@amotin Thanks, that fixed it. I used "draid" now, and as you predicted `zpool status` says `draid1:2d:3c:0s-0`.
`zfs list` shows available space now as expected:
```
NAME                                                USED  AVAIL  REFER  MOUNTPOINT
ZfsBackup1                                         1.18M  8.96T   320K  /ZfsBackup1 
```

**Comment by ChristophSchmidpeter** [comment]
> Proper 3-disk draid, if there is such a thing

@amotin Are you saying dRAID with 3 disks is non-sensical, and I should use RAID-Z instead when using 3 disks?

**Reply by amotin** [reply]
Draid allows faster sequential disk rebuild, but for 3-disk pool without spare it is likely not a priority.  Even faster draid could do embedded spare activation, but again not with only 3 disks.  Also after either completed it starts regular scrub process, which will take just as much time as would regular disk replacement on RAIDZ, just while pool redundancy is temporary restored.  Same time RAIDZ is simpler, known for a long time and has slightly better space efficiency on small objects.
